### ISSUE TITLE
Xviewer shows non-existing black line in svg

### ISSUE THREAD

**Issue by graue70** [linuxmint/xviewer#131]
```
 * Xviewer version: 2.0.2
 * Distribution: Mint 19.1
```

**Issue**

When opening a certain svg (attached below) Xviewer shows a weird black line but Firefox and Inkscape do not show this line. Compare these two screenshots (first Xviewer, second Firefox):
![person_xviewer](https://user-images.githubusercontent.com/23035329/108406645-fa604600-7222-11eb-9640-34c5a757ea94.png)
![person_firefox](https://user-images.githubusercontent.com/23035329/108406660-ff24fa00-7222-11eb-953d-47a132e2bb4f.png)

**Steps to reproduce**

Open the attached svg in Xviewer, open it in Firefox and compare.

**Expected behaviour**

I expect an svg to look the same independent of the viewer that displays it.

**Other information**

Can anybody please check whether the problem is still there in a newer version of Xviewer?

The preview image in Nemo 4.0.6 also incorrectly shows the black line. Does this mean the problem lies somewhere else?

Since I cannot attach the svg, I include its code here (just copy & paste into a file with svg extension):
```svg
<?xml version="1.0" encoding="UTF-8"?><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" fill-opacity="1" color-rendering="auto" color-interpolation="auto" text-rendering="auto" stroke="black" stroke-linecap="square" width="98" stroke-miterlimit="10" shape-rendering="auto" stroke-opacity="1" fill="black" stroke-dasharray="none" font-weight="normal" stroke-width="1" height="118" font-family="'Dialog'" font-style="normal" stroke-linejoin="miter" font-size="12px" stroke-dashoffset="0" image-rendering="auto">
  <!--Generated by ySVG 2.5-->
  <defs id="genericDefs"/>
  <g>
    <defs id="defs1">
      <clipPath clipPathUnits="userSpaceOnUse" id="clipPath1">
        <path d="M0 0 L98 0 L98 118 L0 118 L0 0 Z"/>
      </clipPath>
      <clipPath clipPathUnits="userSpaceOnUse" id="clipPath2">
        <path d="M394 349 L492 349 L492 467 L394 467 L394 349 Z"/>
      </clipPath>
      <clipPath clipPathUnits="userSpaceOnUse" id="clipPath3">
        <path d="M-20.8855 -18.6087 L77.8855 -18.6087 L77.8855 91.0427 L-20.8855 91.0427 L-20.8855 -18.6087 Z"/>
      </clipPath>
    </defs>
    <g fill="white" text-rendering="geometricPrecision" shape-rendering="geometricPrecision" transform="translate(-394,-349)" stroke="white">
      <rect x="394" width="98" height="118" y="349" clip-path="url(#clipPath2)" stroke="none"/>
    </g>
    <g text-rendering="geometricPrecision" shape-rendering="geometricPrecision" transform="matrix(0.9922,0,0,1.0761,20.7225,20.0255)">
      <g clip-path="url(#clipPath3)">
        <svg xml:space="preserve" opacity="1" writing-mode="lr-tb" stop-color="rgb(0, 0, 0)" shape-rendering="auto" glyph-orientation-horizontal="0deg" color-profile="auto" lighting-color="rgb(255, 255, 255)" color="rgb(0, 0, 0)" font-weight="400" alignment-baseline="auto" font-style="normal" version="1.1" color-interpolation-filters="linearrgb" text-anchor="start" stroke-linecap="butt" color-interpolation="srgb" font-variant="normal" word-spacing="normal" fill-opacity="1" text-rendering="auto" clip-path="none" text-decoration="none" letter-spacing="normal" viewBox="0 0 57 65" glyph-orientation-vertical="auto" display="inline" font-size-adjust="none" overflow="hidden" fill="rgb(0, 0, 0)" font-stretch="normal" stroke-dasharray="none" id="svg1.Ebene_1" stroke-miterlimit="4" stop-opacity="1" color-rendering="auto" font-size="12" pointer-events="visiblepainted" mask="none" direction="ltr" baseline-shift="baseline" enable-background="new 0 0 57 65" fill-rule="nonzero" image-rendering="auto" stroke-dashoffset="0" width="57px" marker-end="none" clip="auto" cursor="auto" stroke="none" filter="none" visibility="visible" kerning="auto" stroke-width="1" font-family="&quot;Arial&quot;,&quot;Helvetica&quot;,sans-serif" flood-opacity="1" clip-rule="nonzero" src="none" height="65px" unicode-bidi="normal" stroke-linejoin="miter" stroke-opacity="1" flood-color="rgb(0, 0, 0)" dominant-baseline="auto" marker-start="none" x="0px" marker-mid="none" y="0px">
<g>
	
		<linearGradient gradientTransform="matrix(1 0 0 1 0.0801 -771.6914)" x1="26.5796" x2="27.8207" gradientUnits="userSpaceOnUse" y1="796.6533" y2="826.4517" id="svg1.SVGID_1_">
		<stop offset="0.2711" style="stop-color:#FFAB4F"/>
		<stop offset="1" style="stop-color:#FFD28F"/>
	</linearGradient>
	<path fill="url(#svg1.SVGID_1_)" d="M49.529,52.225c-4.396-4.396-10.951-5.884-12.063-6.109   V38.8H19.278c0,0,0.038,6.903,0,6.868c0,0-6.874,0.997-12.308,6.432C1.378,57.691,0.5,63.77,0.5,63.77   c0,1.937,1.575,3.492,3.523,3.492h48.51c1.947,0,3.521-1.558,3.521-3.492C56.055,63.768,54.211,56.906,49.529,52.225z" stroke-miterlimit="10" stroke="#ED9135"/>
	<path fill="#ECECEC" id="svg1.body_9_" d="M0.5,63.768c0,1.938,1.575,3.494,3.523,3.494h48.51   c1.947,0,3.521-1.559,3.521-3.494c0,0-1.844-6.861-6.525-11.543c-4.815-4.814-11.244-6.146-11.244-6.146   c-1.771,1.655-5.61,3.802-10.063,3.802c-4.453,0-8.292-2.146-10.063-3.802c0,0-5.755,0.586-11.189,6.021   C1.378,57.689,0.5,63.768,0.5,63.768z" stroke-miterlimit="10" stroke="#9B9B9B"/>
	<path fill="#2068A3" d="M28.106,34.486c-8.112,0-12.688,4.313-12.688,10.438c0,7.422,12.688,10.438,12.688,10.438   s14.688-3.016,14.688-10.438C42.793,39.75,36.215,34.486,28.106,34.486z M26.288,54.051c0,0-7.135-2.093-8.805-7.201   c-0.222-0.682,0.147-1.156,0.795-1.522V38.8h20.188v6.663c0.235,0.352,1.109,0.737,1.229,1.387   C40.445,50.917,26.288,54.051,26.288,54.051z" stroke="#2068A3"/>
	<path fill="#5491CF" d="M49.529,52.225c-2.239-2.24-5.041-3.724-7.396-4.67   c-2.854,5.51-14.027,7.807-14.027,7.807s-10.472-2.489-12.387-8.514c-2.439,0.771-5.787,2.287-8.749,5.25   c-5.592,5.592-6.47,11.67-6.47,11.67c0,1.937,1.575,3.492,3.523,3.492h48.51c1.948,0,3.522-1.558,3.522-3.492   C56.055,63.768,54.211,56.906,49.529,52.225z" stroke-miterlimit="10" stroke="#2068A3"/>
	<path fill="#5491CF" d="M13.404,45.173c1.15-1.81,2.039-3.832,3.332-5.397c-0.514,1.027-1.669,4.084-1.669,5.148   c0,5.186,10.366,9.079,14.688,10.438c-3.472,1.627-9.134-1.498-11.34-2.365c-3.601-1.419-4.071-3.063-5.89-4.857   C12.523,48.135,12.878,46,13.404,45.173z" stroke="#2068A3"/>
	<path fill="#5491CF" d="M45.777,44.924c-1.317-1.568-5.11-9.424-6.604-6.617c0.516,1.025,3.617,3.693,3.617,6.617   c0,5.186-10.266,8.576-16.693,9.145c1.429,4.934,11.367,1.293,13.804-0.316c3.563-2.354,4.557-5.133,7.852-3.705   C47.754,50.045,48.006,47.574,45.777,44.924z" stroke="#2068A3"/>
	<path fill="none" d="M30.777,55.167c0.357,0.836-0.153,1.983-0.352,2.813   c-0.256,1.084-0.072,2.104,0.102,3.186c0.164,1.02,0.156,2.107,0.25,3.167c0.082,0.916,0.482,1.849,0.357,2.75" stroke="#2068A3" stroke-linecap="round"/>
	<path fill="none" d="M23.695,54.417c-0.508,0.584-0.476,2.209-0.398,3   c0.116,1.183,0.456,2.099,0.333,3.333c-0.192,1.943,0.154,4.479-0.436,6.333" stroke="#2068A3" stroke-linecap="round"/>
	
		<radialGradient gradientTransform="matrix(1 0 0 1 0.0801 -771.6914)" gradientUnits="userSpaceOnUse" r="23.4236" cx="27.8228" id="svg1.face_x5F_white_1_" cy="798.418" fx="23.2533" fy="795.9283">
		<stop offset="0" style="stop-color:#FFD28F"/>
		<stop offset="1" style="stop-color:#FFAB4F"/>
	</radialGradient>
	<path fill="url(#svg1.face_x5F_white_1_)" id="svg1.face_x5F_white_3_" d="M43.676,24.357   c0.086,10.2-6.738,18.52-15.245,18.586c-8.504,0.068-15.468-8.146-15.554-18.344C12.794,14.4,19.618,6.079,28.123,6.012   C36.627,5.945,43.59,14.158,43.676,24.357z" stroke-miterlimit="10" stroke="#ED9135"/>
	
		<linearGradient gradientTransform="matrix(0.275 0 0 -0.2733 -589.5001 -1032.9376)" x1="2236.3486" x2="2259.9785" gradientUnits="userSpaceOnUse" y1="-3813.3281" y2="-3906.7793" id="svg1.face_highlight_2_">
		<stop offset="0" style="stop-color:#FFFFFF;stop-opacity:0.72"/>
		<stop offset="1" style="stop-color:#FFFFFF;stop-opacity:0.2067"/>
	</linearGradient>
	<path fill="url(#svg1.face_highlight_2_)" id="svg1.face_highlight_1_" d="M27.958,7.333c-6.035,0.047-10.747,4.493-12.787,10.386   c-0.664,1.919-0.294,4.043,0.98,5.629c2.73,3.398,5.729,6.283,9.461,8.088c3.137,1.518,7.535,2.384,11.893,1.247   c2.274-0.592,3.988-2.459,4.375-4.766c0.181-1.094,0.293-2.289,0.283-3.553C42.083,14.952,36.271,7.268,27.958,7.333z"/>
	<path fill="#FFB380" d="M32.215,10.938c0,0,5.688,2.75,7.688,8.125c2.104,5.652,4.123,8.232,4.188,8   c1.875-6.794,1.063-21.438-10.17-21.587C13.467-2.187,8.342,17.438,10.029,25.126c1.078,4.911,2.234,6.686,3.938,8.08   C13.966,33.205,15.028,18.563,32.215,10.938z" stroke="#953606"/>
	<path fill="#CC9869" d="M32.215,10.938   c0,0,5.688,2.75,7.688,8.125c2.104,5.652,4.123,8.232,4.188,8c1.875-6.794,1.063-21.438-10.17-21.587   C13.467-2.187,8.342,17.438,10.029,25.126c1.078,4.911,2.234,6.686,3.938,8.08C13.966,33.205,15.028,18.563,32.215,10.938z" stroke="#99724F" stroke-linecap="round" stroke-linejoin="round"/>
</g>
</svg>
      </g>
    </g>
  </g>
</svg>
```

**Comment by programmer-ceds** [comment]
No black line in xviewer 2.8.1 (plus PR 129 included). Mint 20.1

(You could have zipped the svg and posted the zip file)

**Comment by graue70** [comment]
> No black line in xviewer 2.8.1 (plus PR 129 included). Mint 20.1

Cool, thanks! It's not dependent on #129, is it? In that case I will close this issue.

**Comment by programmer-ceds** [comment]
I've just downloaded and built the latest master version (2.8.3) on another Mint 20.1 system and that also doesn't show the black line. But then I built V2.0.2 and this didn't show the black line either. Could it be related to Mint 19.1 - seems unlikely. Graphics hardware issue? - again seems unlikely. Anyway I can't reproduce it on my PC or laptop.

**Comment by graue70** [comment]
> I've just downloaded and built the latest master version (2.8.3) on another Mint 20.1 system and that also doesn't show the black line. But then I built V2.0.2 and this didn't show the black line either. Could it be related to Mint 19.1 - seems unlikely. Graphics hardware issue? - again seems unlikely. Anyway I can't reproduce it on my PC or laptop.

Like I said: The image preview in Nemo 4.0.6 also shows the black line. I don't know enough about the Mint ecosystem to know what that means though.

**Comment by programmer-ceds** [comment]
I also get the same results with inkscape and pix - that is to say that the code you posted displays the face without the black line.

Looking at the code my understanding is that the image buffer is loaded (rendered) by a call to the GTK+ function rsvg_handle_get_pixbuf() - in which case the error that you are seeing would lie at a lower level than xviewer, rather than in xviewer itself.

**Comment by graue70** [comment]
> What do you see if you type the following in a terminal window?
> 
> `dpkg -s librsvg2-2|grep '^Version'`

`Version: 2.40.20-2ubuntu0.2`

**Comment by programmer-ceds** [comment]
I'm using Mint 20.1 and the equivalent output on my system is:

2.48.9-1ubuntu0.20.04.1

Clearly many versions ahead. The list of changes is extensive although I couldn't spot anything that looked like the problem that you are having. I installed Mint 19.1 Cinnamon in VirtualBox and see the black line in xviewer, pix, GIMP and the thumbnail (pix might have been updated on your system - I just installed the original 19.1 ISO)

It therefore looks like it may well be a problem that has been resolved and anyway was at a lower level than xviewer.

Are you able (or want) to update to 20.1 - there are instructions for this on the Mint website.

**Comment by graue70** [comment]
Ubuntu 18.04 (which Linux Mint 19 is based on) [uses my version of librsvg](https://packages.ubuntu.com/bionic/librsvg2-2) so I guess it would be useful to check whether it has similar problems. If so, we should report the problem upstream in order to get it fixed there. After all, both Ubuntu 18.04 and Linux Mint 19 are supported for two full years.

**Comment by programmer-ceds** [comment]
I'll leave you to chase that - it may be selfish but using Mint 20.1 I don't have the problem - and the same solution of updating is available to others. If you do get it fixed it would be worth checking to see if it also fixes issue #137. Good luck.

**Comment by graue70** [comment]
I'm closing this because the problem is located in some ubuntu or debian libraries and not in xviewer. Also, it has been fixed already (as discussed above).
